app/src/main/java/com/example/home/stream/StreamFragment.java

```diff
@@ -0,0 +1,133 @@
+
+
+    public StreamFragment(String mNoOfBedrooms, String mPropertyTypeText, Integer mZipCode) {
+        this.mNoOfBedrooms = mNoOfBedrooms;
+        this.mPropertyTypeText = mPropertyTypeText;
+        this.mZipCode = mZipCode;
+        // Required empty public constructor
+    }
+
+    // TODO: Rename and change types and number of parameters
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_stream, container, false);
+
+        return view;
+
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        streamRecyclerView = view.findViewById(R.id.HomesRecyclerView);
+        StreamLinearLayout = view.findViewById(R.id.StreamLinearLayout);
+        //initializes list of homes and adapter
+        mHome = new ArrayList<>();
+        //recycler view setup
+        adapter = new HomeAdapter(getContext(), mHome);
+        streamRecyclerView.setAdapter(adapter);
+        streamRecyclerView.setLayoutManager(new LinearLayoutManager(StreamLinearLayout.getContext()));
+        AsyncHttpClient client = new AsyncHttpClient();
+        RequestHeaders headers = new RequestHeaders();
+        headers.put("apikey", getContext().getString(R.string.attomdata_api_key));
+        RequestParams params = new RequestParams();
+
+        params.put(LATITUDE_PARAM, "47.628891");
+        params.put(LONGITUDE_PARAM, "-122.341408");
+        params.put(RADIUS_PARAM, "1");
+        params.put(MIN_BEDS_PARAM, mNoOfBedrooms);
+        params.put(MAX_BEDS_PARAM, mNoOfBedrooms);
+        params.put(PROPERTY_TYPE_PARAM, mPropertyTypeText);
+        client.get("https://api.gateway.attomdata.com/propertyapi/v1.0.0/property/snapshot?", headers, params, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Headers headers, JSON json) {
+                // Access a JSON array response with `json.jsonArray`
+                //Log.d("DEBUG ARRAY", json.jsonArray.toString());
+                populateHomes(json.jsonObject);
+                adapter.notifyDataSetChanged();
+                Log.d("Home size", String.valueOf(mHome.size()));
+                // Access a JSON object response with `json.jsonObject`
+                Log.d("DEBUG OBJECT", json.jsonObject.toString());
+            }
+            @Override
+            public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
+                Log.e("ERROR:", response.toString());
+
+            }
+        });
```

Comment:
Let's have a separate function that does this. You can call it Fetch Houses or something like that